pettingzoo/utils/env.py

```diff
@@ -73,7 +73,6 @@ def reset(
         """Resets the environment to a starting state."""
         raise NotImplementedError
 
-    # TODO: Remove `Optional` type below
     def observe(self, agent: AgentID) -> ObsType | None:
```

Comment:
I feel like there was a reason for this being none, I think in the PR that added AgentID (can check the git blame here) they said that there were some examples that it returned none or something? Except that doesn't make sense, and seems like a bug. 

Maybe it's that some envs just update a self.observations dict or something? But yeah it should just be at least type hinted to show people that you are intended to return an ObsType.